### PR TITLE
Use IntersectionObserver to lazyload comments

### PR DESCRIPTION
**Pull request by jiangtj**
## PR Checklist 

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [x] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the new behavior?

- Link to demo site with this changes: https://jiangtj.gitlab.io/next-test/gemini/post/hello-world/

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by stevenjoezhang** [comment]
https://github.com/theme-next/hexo-theme-next/pull/1152#issuecomment-527824894

**Comment by jiangtj** [comment]
https://github.com/theme-next/hexo-next-utteranc/blob/6f841a67c9d12f1453be8269abb6307862542753/index.js#L21  It only has a <script>. Although it can be changed, it is a bit annoying. Anyway, just lazy loading doesn't work, no other effect.

File: layout/_third-party/comments/valine.swig [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+NexT.utils.loadComments(document.querySelector('#valine-comments'), () => {
   NexT.utils.getScript('{{ valine_uri }}', () => {
     var GUEST = ['nick', 'mail', 'link'];
     var guest = '{{ theme.valine.guest_info }}';
     guest = guest.split(',').filter(item => {
       return GUEST.includes(item);
     });
     new Valine({
-      el: '#comments',
+      el: '#valine-comments',
```
Comment (sli1989) [inline layout/_third-party/comments/valine.swig]: this makes the hyperlink doesn't working.
https://github.com/theme-next/hexo-theme-next/blob/34ffcee4a73155133b1b0049dcaf816a00703d00/scripts/filters/comment/valine.js#L28

**Comment by sli1989** [comment]
有没有可能都变成 `id="comments"` 这样的超链接。不同评论不同跳转tag感觉也不是很好。

**Pull request merged**